src/navigation_controller.ts

```diff
@@ -139,29 +138,43 @@ export class NavigationController {
   }
 
   /**
-   * Sets whether the navigation controller should automatically intercept tab,
-   * enter, and escape navigation events to ensure that the user needs to
-   * explicitly enter and exit the Blockly environment for keyboard shortcuts to
-   * be enabled.
+   * Sets whether the navigation controller has focus. This has no side effect
+   * unless auto navigation is enabled via setHasAutoNavigationEnabled.
    *
-   * This mode has no effect unless setHasFocus is correctly used to communicate
-   * focus gained and lost contexts.
+   * @param isFocused whether the environment has browser focus.
+   */
+  setHasFocus(isFocused: boolean) {
+    this.hasNavigationFocus = isFocused;
+  }
+
+  /**
+   * Determines whether keyboard navigation should be allowed based on the
+   * current state of the workspace.
+   *
+   * A return value of 'true' generally indicates that the workspace both has
+   * enabled keyboard navigation and is currently in a state (e.g. focus) that
+   * can support keyboard navigation.
    *
-   * @param autoNavEnabled whether the controller should automatically
-   *     intercept navigation events.
+   * @param workspace the workspace in which keyboard navigation may be allowed.
+   * @returns whether keyboard navigation is currently allowed.
    */
-  setHasAutoNavigationEnabled(autoNavEnabled: boolean) {
-    this.isAutoNavigationEnabled = autoNavEnabled;
+  private isKeyboardNavigationCurrentlyAllowed(workspace: WorkspaceSvg) {
```

Comment:
TIL you not only have [Java](https://www.reddit.com/r/ProgrammerHumor/comments/xb18vc/java_naming_conventions/?rdt=49176) readability but you are a Java readability reviewer…

Would `canNavigate` or, if you think that too inspecific, `canCurrentlyNavigate` suffice?

Reply:
I went into more details on 'currently' and 'can' in my other comment, but I think I lean toward ``canCurrentlyNavigate`` unless you have any suggestions that for clearer names?